replab_init.m

```diff
@@ -162,6 +162,31 @@ function replab_init(verbose)
         replab.settings.replabPath(pathStr);
     end
 
+
+    %% Verifying that the nlinfit function is available
+
```

Comment:
@jdbancal : that's the main problem of the new approach. It requires Levenberg-Marquardt fitting the algorithm convergence. It seems ridiculous to add a dependency on the Matlab stats toolbox just for that.

There is plenty of Octave/Matlab code around to do that, and we don't need all the bells and whistles. For our fit function, we can compute an explicit Jacobian, etc.. and just focus on the method that we need.

A possible inspiration is http://people.duke.edu/~hpgavin/ce281/lm.pdf

Reply:
Hmm, @denisrosset can you precise where do we need this function (in which context)? And what purpose does it fill?

Reply:
It fits the approximation error for the iterative averaging procedure over compact groups, as to estimate the noise floor and perform convergence checks.

Reply:
What I see is that this method is used only in the file: https://github.com/replab/replab/blob/7cd850cb156fef569fef9d2bc98dfba115ef9cbf/src/%2Breplab/%2Bequivariant/ForCompactGroup.m#L57
It is used to fit the function
![image](https://user-images.githubusercontent.com/5440791/73942223-fc44b900-48ee-11ea-931c-489786aa645a.png)
with parameters b1,b2,b3 to some data vectors x and log(y).

I imagine that the reason we rely on the `nlinfit` method is because the modelled function is very nonlinear.

Reply:
Yes! also that least squares fitting doesn't have a complicated Hessian (it only depends on the first order derivatives of the fit function)

Thus, if we continue to use `nlinfit`, I expect that it's a small project (a day?) to write a special purpose Levenberg-Marquardt implementation for RepLAB. It's a shame that the available code is either 1) under the GPL (Octave) 2) has distribution terms that prohibit use with Octave (Matlab Central)

Reply:
BTW, the function is simply the sum of an exponential term and a constant; the weird form is to be compatible with a log10 plot.

Reply:
Thanks for your feedback!
Well, I guess that the log10 makes the plot nice, but also affects the fitting: least squares computed with log10 would not be minimal without the log10. But I guess we should indeed use log10 here as the behavior is probably better with the log10 as it corresponds mostly to fitting two lines. It's a pitty we can't just do an svd, here, would be even simpler :-)